crates/bevy_reflect/src/type_registry.rs

```diff
@@ -803,4 +892,24 @@ mod test {
             }
         }
     }
+
+    #[test]
+    fn type_data_iter() {
```

Comment:
We should probably add a new test function to assert that iter_mut is working properly.

Reply:
Maybe the new function isn't needed, just testing this inside of this function will probably work

Reply:
I went ahead and just did a separate test case anyways :)